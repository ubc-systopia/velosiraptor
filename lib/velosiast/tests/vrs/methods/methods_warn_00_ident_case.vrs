// upper case
fn FOOBAR(foo: int, bar: int) -> int
{
    1
}

// camel case
fn FooBar(foo: int, bar: int) -> int
{
    1
}

// upper case param
fn foo(FOO: int, bar: int) -> int
{
    1
}

// camel case param
fn foo(FooBar: int, bar: int) -> int
{
    1
}