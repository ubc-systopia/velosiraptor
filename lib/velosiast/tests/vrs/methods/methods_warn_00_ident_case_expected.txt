fn FOOBAR(foo: int, bar: int) -> int
  requires true;
{
  0x1
}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: identifier `FOOBAR` should have a snake case name
     --> tests/vrs/methods/methods_warn_00_ident_case.vrs:2:4
      |
    2 |  fn FOOBAR(foo: int, bar: int) -> int
      |     ^^^^^^ convert the identifier to lowercase (notice the snake_case): `foobar`
warning: generated 1 warnings


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

fn FooBar(foo: int, bar: int) -> int
  requires true;
{
  0x1
}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: identifier `FooBar` should have a snake case name
     --> tests/vrs/methods/methods_warn_00_ident_case.vrs:8:4
      |
    8 |  fn FooBar(foo: int, bar: int) -> int
      |     ^^^^^^ convert the identifier to lowercase (notice the snake_case): `foobar`
warning: generated 1 warnings


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

fn foo(FOO: int, bar: int) -> int
  requires true;
{
  0x1
}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: identifier `FOO` should have a snake case name
     --> tests/vrs/methods/methods_warn_00_ident_case.vrs:14:8
      |
   14 |  fn foo(FOO: int, bar: int) -> int
      |         ^^^ convert the identifier to lowercase (notice the snake_case): `foo`
warning: generated 1 warnings


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

fn foo(FooBar: int, bar: int) -> int
  requires true;
{
  0x1
}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: identifier `FooBar` should have a snake case name
     --> tests/vrs/methods/methods_warn_00_ident_case.vrs:20:8
      |
   20 |  fn foo(FooBar: int, bar: int) -> int
      |         ^^^^^^ convert the identifier to lowercase (notice the snake_case): `foobar`
warning: generated 1 warnings


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
